### PR TITLE
Add and remove PR reviewers

### LINKED ISSUE #1102: Add reviewers to PR
Dynamically add to manage tray for PRs-only

### PR DESCRIPTION
**Pull request by rnystrom**
Fixes #1102

pr-reviews

### CODE REVIEW

File: Classes/Issues/GithubClient+Issues.swift [hunk 1]
```diff
@@ -396,17 +396,42 @@ extension GithubClient {
...
+    }
+
+    func addPeople(
+        type: AddPeopleType,
         previous: IssueResult,
         owner: String,
         repo: String,
         number: Int,
-        assignees: [IssueAssigneeViewModel]
+        people: [IssueAssigneeViewModel]
```
Comment (Sherlouk) [inline Classes/Issues/GithubClient+Issues.swift]: Now now, let's not exclude robots! Inclusivity!

File: Classes/Issues/IssueResult.swift [hunk 2]
```diff
@@ -120,6 +120,29 @@ struct IssueResult: Cachable {
         )
     }
 
+    func withReviewers(
```
Comment (Sherlouk) [inline Classes/Issues/IssueResult.swift]: Out of curiosity, why choose this pattern over using a class and mutating directly?

Comment (rnystrom) [reply Classes/Issues/IssueResult.swift]: Mutations = bad:

- Thread safety
- Race conditions
- Complication

Immutability requires a little bit more setup, but once the model is made you can go wild w/ it and rest assured that nothing will break.

**Comment by BasThomas** [comment]
Neat! Re layout: how will this work when adding another button? Would it then be 3 on the first row, 2 on the bottom or something? Wondering how you imagined this scaling.

File: Classes/Issues/Managing/IssueManagingActionCell.swift [hunk 3]
```diff
@@ -28,7 +28,7 @@ final class IssueManagingActionCell: UICollectionViewCell, ListBindable {
         imageView.snp.makeConstraints { make in
             make.size.equalTo(Styles.Sizes.buttonIcon)
             make.centerX.equalTo(contentView)
-            make.centerY.equalTo(contentView).offset(-7)
+            make.centerY.equalTo(contentView).offset(-10)
```
Comment (BasThomas) [inline Classes/Issues/Managing/IssueManagingActionCell.swift]: Why's this?

File: Classes/Issues/Managing/IssueManagingExpansionModel.swift [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    let expanded: Bool
+
+    init(expanded: Bool) {
+        self.expanded = expanded
+    }
+
+    // MARK: ListDiffable
+
+    func diffIdentifier() -> NSObjectProtocol {
+        return "managing-model" as NSObjectProtocol
```
Comment (BasThomas) [inline Classes/Issues/Managing/IssueManagingExpansionModel.swift]: Shouldn't this return a unique key?

File: Classes/Issues/Managing/IssueManagingModel.swift [hunk 5]
```diff
@@ -11,22 +11,26 @@ import IGListKit
...
+        self.objectId = objectId
+        self.pullRequest = pullRequest
     }
 
     // MARK: ListDiffable
 
     func diffIdentifier() -> NSObjectProtocol {
-        return "managing-model" as NSObjectProtocol
+        // should only ever be one
+        return "managing_model" as NSObjectProtocol
```
Comment (BasThomas) [inline Classes/Issues/Managing/IssueManagingModel.swift]: Why the underscore here and the dash [here](https://github.com/rnystrom/GitHawk/pull/1122/files#diff-63c174961e625ddb07bf11ddd1035adbR23)?

Also, if there can be only one, why does this need to conform to `ListDiffable`?

File: Classes/Issues/Status/IssueStatusSectionController.swift [hunk 6]
```diff
@@ -13,7 +13,7 @@ final class IssueStatusSectionController: ListGenericSectionController<IssueStat
 
     override init() {
         super.init()
-        inset = Styles.Sizes.listInsetTight
+        inset = UIEdgeInsets(top: 4, left: 0, bottom: 0, right: 0)
```
Comment (BasThomas) [inline Classes/Issues/Status/IssueStatusSectionController.swift]: Why did you change this / is this not encapsulated in the `Sizes` anymore?

File: Classes/People/PeopleViewController.swift [hunk 7]
```diff
@@ -33,7 +37,7 @@ final class PeopleViewController: UITableViewController {
         super.viewDidLoad()
 
         tableView.refreshControl = feedRefresh.refreshControl
-        feedRefresh.refreshControl.addTarget(self, action: #selector(LabelsViewController.onRefresh), for: .valueChanged)
+        feedRefresh.refreshControl.addTarget(self, action: #selector(onRefresh), for: .valueChanged)
```
Comment (BasThomas) [inline Classes/People/PeopleViewController.swift]: 👌

**Comment by rnystrom** [comment]
@BasThomas it’s setup to “line break” at 4 items per row 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by BasThomas** [comment]
On all devices?

File: Classes/Issues/Managing/IssueManagingActionCell.swift (hunk 3 above)
Comment (rnystrom) [reply Classes/Issues/Managing/IssueManagingActionCell.swift]: Nudging the vertical spacing a little

File: Classes/Issues/Managing/IssueManagingExpansionModel.swift (hunk 4 above)
Comment (rnystrom) [reply Classes/Issues/Managing/IssueManagingExpansionModel.swift]: There's only one expansion button (the one that says "Manage")

File: Classes/Issues/Managing/IssueManagingModel.swift (hunk 5 above)
Comment (rnystrom) [reply Classes/Issues/Managing/IssueManagingModel.swift]: Can't use in IGListKit if not `ListDiffable` 😛

**Comment by rnystrom** [comment]
> On all devices?

Atm ya
